client/transaction-pool/src/lib.rs

```diff
@@ -563,160 +570,232 @@ async fn prune_known_txs_for_block<Block: BlockT, Api: graph::ChainApi<Block = B
 	hashes
 }
 
-impl<PoolApi, Block> MaintainedTransactionPool for BasicPool<PoolApi, Block>
+impl<PoolApi, Block> BasicPool<PoolApi, Block>
 where
 	Block: BlockT,
 	PoolApi: 'static + graph::ChainApi<Block = Block>,
 {
-	fn maintain(&self, event: ChainEvent<Self::Block>) -> Pin<Box<dyn Future<Output = ()> + Send>> {
-		match event {
-			ChainEvent::NewBestBlock { hash, tree_route } => {
-				let pool = self.pool.clone();
-				let api = self.api.clone();
+	/// enactment_state getter, intended for tests only
+	#[doc(hidden)]
+	pub fn enactment_state(&self) -> Arc<Mutex<EnactmentState<Block>>> {
+		self.enactment_state.clone()
+	}
+}
 
-				let id = BlockId::hash(hash);
-				let block_number = match api.block_id_to_number(&id) {
-					Ok(Some(number)) => number,
-					_ => {
-						log::trace!(
-							target: "txpool",
-							"Skipping chain event - no number for that block {:?}",
-							id,
-						);
-						return Box::pin(ready(()))
-					},
-				};
+impl<PoolApi, Block> BasicPool<PoolApi, Block>
+where
+	Block: BlockT,
+	PoolApi: 'static + graph::ChainApi<Block = Block>,
+{
+	fn handle_enactment(
```

Comment:
All of this here is just a verbatim copy of the old code?

Reply:
The view with whitespaces ignored is [better](https://github.com/paritytech/substrate/pull/12305/files?diff=split&w=1).

Reply:
Well, some additional changes were made: [64a2ec0](https://github.com/paritytech/substrate/pull/12305/commits/64a2ec0015416423073dc2ebab9cfb745633855d),

as requested by @davxy in https://github.com/paritytech/substrate/pull/12305#discussion_r991303896